src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -411,16 +416,24 @@ export const DatasetBuilderContents = ({ onStateChange }: { onStateChange: OnSta
   ]);
 };
 
-type DatasetBuilderState = 'homepage' | 'cohort-editor' | 'concept-selector' | 'concept-set-creator';
-type OnStateChangeType = (state: DatasetBuilderState) => void;
-
 interface DatasetBuilderProps {
   datasetId: string;
+  initialState?: DatasetBuilderState;
+}
+
+export class HomepageState implements DatasetBuilderState {
```

Comment:
hmmm.... I'm not sure what this being a class buys us... A simple default DatasetBuilderState 'string' value of 'homepage' seems sufficient.

Reply:
I poked around the CohortEditorState mechanics... and unless I am missing something, that too should work as a simple type-safe usage of useState hook:

`useState<ComplexType>(initialComplexValue)`

Reply:
This seems equivalent type-wise to using a class. Is there a downside to using a class here? Using a class lets us have a constructor so we know the `cohort` field is populated.

Reply:
I ended up changing the class types to object types (interfaces); I'm not convinced one approach is better than the other but so far we're not using class types in the code base for this type of object.